hnn_core/network.py

```diff
@@ -219,47 +305,408 @@ def __repr__(self):
+            a function of the space_constant
+        cell_specific : bool
+            Whether each cell has unique connection parameters (default: True)
+            or all cells have common connections to a global (single) drive.
+
+        Returns
+        -------
+        drive_conn_by_cell : dict
+            Keys are target_populations, each item is itself a dict with keys
+            relevant to how the drive connects to the rest of the network.
+        src_gid_ran : range
+            For convenience, return back the range of GIDs associated with
+            all the driving units (they become _ArtificialCells later)
+
+        Note
+        ----
+        drive_conn : dict
+            'target_gids': range of target cell GIDs
+            'target_type': target cell type (e.g. 'L2_basket')
+            'src_gids': range of source (artificial) cell GIDs
+            'location': 'distal' or 'proximal'
+            'ampa' and 'nmda': dict
+                'A_weight': synaptic weight
+                'A_delay':  synaptic delay (used with space constant)
+                'lamtha': space constant
+        """
+        drive_conn_by_cell = dict()
+        src_gid_ran_begin = self._n_gids
+
+        if cell_specific:
+            for cellname in target_populations:
+                drive_conn = dict()  # NB created inside target_pop-loop
+                drive_conn['location'] = location
+
+                drive_conn['target_gids'] = self.gid_ranges[cellname]
+                # NB list! This is used later in _parnet_connect
+                drive_conn['target_type'] = cellname
+                drive_conn['src_gids'] = range(
+                    self._n_gids,
+                    self._n_gids + len(drive_conn['target_gids']))
+                self._n_gids += len(drive_conn['target_gids'])
+                drive_conn_by_cell[cellname] = drive_conn
+        else:
+            drive_conn = dict()
+            drive_conn['location'] = location
+
+            # NB list! This is used later in _parnet_connect
+            drive_conn['src_gids'] = [self._n_gids]
+            self._n_gids += 1
+
+            drive_conn['target_gids'] = list()  # fill in below
+            for cellname in target_populations:
+                drive_conn['target_gids'].extend(self.gid_ranges[cellname])
+                drive_conn['target_type'] = cellname
+                drive_conn_by_cell[cellname] = drive_conn
+
+        for cellname in target_populations:
+            for receptor, weights in weights_by_receptor.items():
+                drive_conn_by_cell[cellname][receptor] = dict()
+                if cellname in weights:
+                    drive_conn_by_cell[cellname][receptor][
+                        'lamtha'] = space_constant
+                    if isinstance(synaptic_delays, float):
+                        drive_conn_by_cell[cellname][receptor][
+                            'A_delay'] = synaptic_delays
+                    else:
```

Comment:
Maybe `elif isinstance(synaptic delays, dict)`?

Reply:
Made the change, though there's a test to ensure it's either a float or a dict (so not including an `else: raise ValueError` here.